java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -359,11 +357,19 @@ public boolean isAutoScanCheckFiltering() {
   }
 
   /**
-   * Returns the batch mode size as read from configuration. If not value can be found, returns -1L.
+   * Returns the batch mode size as read from configuration. If not value can be found, compute dynamically an ideal value.
    * @return the batch mode size or a default value of -1L.
    */
   public long getBatchModeSizeInKB() {
-    return context.config().getLong(SONAR_BATCH_SIZE_KEY).orElse(-1L);
+    Configuration config = context.config();
+    if (isAutoScan()) {
+      return -1L;
+    }
+    return config.getLong(SONAR_BATCH_SIZE_KEY).orElse(computeIdealBatchSize());
+  }
+
+  private static long computeIdealBatchSize() {
+    return (long) Math.ceil((Runtime.getRuntime().totalMemory() * 0.05) / 1000L);
```

Comment:
Might be worth mentioning the rationale behind this computation :)

([for reference](https://www.reddit.com/r/ProgrammerHumor/comments/9srvyc/code_from_quake_iii_arena_including_exact/))